### PR TITLE
[4.4 cherry pick] Add a new value `neo4j.passwordFromSecret` (#122)

### PR DESCRIPTION
**Pull request by ojhughes**
* add passwordFromSecret value
* update integration tests
* Add warning message when using neo4j.password
* Update NOTES.txt to output command to retrieve password when using passwordFromSecret
* Add check that secret data uses format 'neo4j/...'

### CODE REVIEW

File: internal/model/release_values.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+}
+type Volumes struct {
+	Data     Data     `yaml:"data,omitempty"`
+	Backups  Backups  `yaml:"backups,omitempty"`
+	Logs     Logs     `yaml:"logs,omitempty"`
+	Metrics  Metrics  `yaml:"metrics,omitempty"`
+	Import   Import   `yaml:"import,omitempty"`
+	Licenses Licenses `yaml:"licenses,omitempty"`
+}
+type NodeSelector struct {
```
Comment (harshitsinghvi22) [inline internal/model/release_values.go]: empty struct ?

File: internal/model/release_values.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+}
+type Annotations struct {
+}
```
Comment (harshitsinghvi22) [inline internal/model/release_values.go]: empty struct ?

File: internal/model/release_values.go [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+	Enabled bool `yaml:"enabled,omitempty"`
+}
+type Ports struct {
+	HTTP   HTTPPort   `yaml:"http,omitempty"`
+	HTTPS  HTTPSPort  `yaml:"https,omitempty"`
+	Bolt   BoltPort   `yaml:"bolt,omitempty"`
+	Backup BackupPort `yaml:"backup,omitempty"`
+}
+type ServiceSelector struct {
+	HelmNeo4JComNeo4JLoadbalancer string `yaml:"helm.neo4j.com/neo4j.loadbalancer,omitempty"`
```
Comment (harshitsinghvi22) [inline internal/model/release_values.go]: can we change the name HelmNeo4JComNeo4JLoadbalancer to something more easy to read?

File: internal/model/release_values.go [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+	ImagePullPolicy string `yaml:"imagePullPolicy,omitempty"`
+	CustomImage     string `yaml:"customImage,omitempty"`
+}
+type Metadata struct {
+	Annotations map[string]string `yaml:"annotations,omitempty"`
+}
+type Statefulset struct {
+	Metadata Metadata `yaml:"metadata,omitempty"`
+}
+type Env struct {
```
Comment (harshitsinghvi22) [inline internal/model/release_values.go]: empty struct ?

File: internal/model/release_values.go [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+}
+type Dynamic struct {
+	StorageClassName string   `yaml:"storageClassName,omitempty"`
+	AccessModes      []string `yaml:"accessModes,omitempty"`
+	Requests         Requests `yaml:"requests,omitempty"`
+}
+type Volume struct {
+	SetOwnerAndGroupWritableFilePermissions bool `yaml:"setOwnerAndGroupWritableFilePermissions,omitempty"`
+}
+type VolumeClaimTemplate struct {
```
Comment (harshitsinghvi22) [inline internal/model/release_values.go:70]: empty struct ?

**Comment by harshitsinghvi22** [comment]
i see this PR is doing two things...adding the new model of values.yaml and support for passwordFromSecret....can we have two PR's two differentiate and it can help with a better review

**Review by harshitsinghvi22 — CHANGES_REQUESTED** [review]

**Comment by ojhughes** [comment]
> i see this PR is doing two things...adding the new model of values.yaml and support for passwordFromSecret....can we have two PR's two differentiate and it can help with a better review

I didn't want to pull the tests improvements but have to as the cherry picked tests depend on them. Otherwise I will have to rewrite those tests

File: internal/model/release_values.go (hunk 4 above)
Comment (ojhughes) [reply internal/model/release_values.go]: they need to be empty as they represent a `map[string] string` YAML entry

File: internal/model/release_values.go (hunk 3 above)
Comment (harshitsinghvi22) [reply internal/model/release_values.go]: type ServiceSelector struct { --> this should be a map[string]string with the below as default value
"helm.neo4j.com/neo4j.loadbalancer": "include"

Comment (harshitsinghvi22) [reply internal/model/release_values.go]: i dont see this serviceSelector in 4.4

File: internal/model/release_values.go (hunk 2 above)
Comment (harshitsinghvi22) [reply internal/model/release_values.go]: i dont see Annotations in 4.4 values.yaml ...which annotations are these ? PodSpec ? statefulset ?

File: internal/model/release_values.go (hunk 4 above)
Comment (harshitsinghvi22) [reply internal/model/release_values.go]: should we declare it like this instead of empty struct ?
type Env map[string]string

**Review by harshitsinghvi22 — APPROVED** [review]

**Pull request merged**